test/foundry/mocks/policy/MockRoyaltyPolicyLAP.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.23;
+
+import { ReentrancyGuardUpgradeable } from "@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol";
+import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+import { GovernableUpgradeable } from "../../../../contracts/governance/GovernableUpgradeable.sol";
+import { RoyaltyPolicyLAP } from "../../../../contracts/modules/royalty/policies/RoyaltyPolicyLAP.sol";
+import { IRoyaltyPolicyLAP } from "../../../../contracts/interfaces/modules/royalty/policies/IRoyaltyPolicyLAP.sol";
+import { Errors } from "../../../../contracts/lib/Errors.sol";
+
+contract MockRoyaltyPolicyLAP is IRoyaltyPolicyLAP, GovernableUpgradeable, ReentrancyGuardUpgradeable, UUPSUpgradeable {
```

Comment:
Why do we need this when other mocks are not used anymore?

Reply:
A lot of License-specific tests fail due to RoyaltyPolicyLAP, which requires allowance, payment, etc. related to commercial licenses. I thought these tests should only deal with licenses, so I used a mock RoyaltyPolicyLAP. 

However, if we deem these tests more as integration tests, then I will use RoyaltyPolicyLAP. It will require dealing with LAP allowance and payments for minting licenses in the tests.

Reply:
MockRoyaltyPolicyLAP is almost a fully functional contract, upgradeable and everything.
Do we need this for a mock?

If we don't want to treat license specific tests as integrations with RoyaltyPolicyLAP (which I think it's very valid, since we want it to work with multiple policies), why don't we use a barebones mock inheriting only from `IRoyaltyPolicy.sol`?

If that's not enough, we should revisit the interface because that would indicate tight integration with RoyaltyPolicyLAP.

If we have mocks, I believe we would benefit from them being very low maintenance.

Reply:
Yes, I will keep it minimal as only inheriting IRoyaltyPolicy.sol, thanks for the note!